File: 17-08-2024/Pair_with_given_sum_in_a_sorted_array.java
// Initial Template for Java

import java.io.*;
import java.lang.*;
import java.util.*;


// } Driver Code Ends
// User function Template for Java

class Solution {

    int countPair(int k, int arr[]) {
        // Complete the function
        int count=0;
        int low =0;
        int high = arr.length-1;
        while(low<high){
            if(arr[low] + arr[high]<k){
                low++;
            }
            else if(arr[low]+arr[high]>k){
                high--;
            }
            else if(arr[low]+arr[high] == k){
                count++;
                low++;
                high--;
            }
        }
        return count;
    }
}


//{ Driver Code Starts.
class Pair_with_given_sum_in_a_sorted_array {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(br.readLine().trim()); // Inputting the testcases
        while (t-- > 0) {
            String[] inputLine = br.readLine().trim().split(" ");
            int k = Integer.parseInt(inputLine[0]);

            inputLine = br.readLine().trim().split(" ");
            int[] arr = new int[inputLine.length];
            for (int i = 0; i < inputLine.length; i++) {
                arr[i] = Integer.parseInt(inputLine[i]);
            }

            Solution obj = new Solution();
            int res = obj.countPair(k, arr);
            System.out.println(res);
        }
    }
}
// } Driver Code Ends

File: 17-08-2024/Pair_with_given_sum_in_a_sorted_array.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends

class Solution {
  public:
    int countPair(int k, vector<int> &arr) {
        // Complete the function
        int count=0;
        int low =0;
        int high = arr.size()-1;
        while(low<high){
            if(arr[low] + arr[high]<k){
                low++;
            }
            else if(arr[low]+arr[high]>k){
                high--;
            }
            else if(arr[low]+arr[high] == k){
                count++;
                low++;
                high--;
            }
        }
        return count;
    }
};

//{ Driver Code Starts.
int main() {

    int t;
    cin >> t;

    while (t--) {
        vector<int> arr;
        int k;
        cin >> k;
        string input;
        getline(cin, input);
        getline(cin, input);

        stringstream ss(input);
        int number;
        while (ss >> number) {
            arr.push_back(number);
        }

        Solution obj;
        cout << obj.countPair(k, arr) << endl;
    }

    return 0;
}
// } Driver Code Ends